### ISSUE TITLE
Analyse fastq from first QC - Illumina and Nanopore

### ISSUE THREAD

**Issue by Alema91** [BU-ISCIII/viralrecon#19]
Viralrecon analysis of the QC samples to compare with the QC results.

**Comment by Alema91** [comment]
Problems with:

- Nanopore
- Using VarSkip amplicons: [Varskip](https://github.com/nebiolabs/VarSkip)

**Comment by Alema91** [comment]
Problem with nanopore samples:

```
-[nf-core/viralrecon] Pipeline completed with errors-
Error executing process > 'NFCORE_VIRALRECON:NANOPORE:ARTIC_MINION (COD_2106_Nanopore_1)'

Caused by:
  Argument of `file` function cannot be empty

Source block:
  def args = task.ext.args   ?: ''
  prefix   = task.ext.prefix ?: "${meta.id}"
  def version  = scheme_version.toString().toLowerCase().replaceAll('v','')
  def fast5    = fast5_dir ? "--fast5-directory $fast5_dir"             : ""
  def summary  = sequencing_summary ? "--sequencing-summary $sequencing_summary" : ""
  def model    = ""
  if (args.tokenize().contains('--medaka')) {
          fast5   = ""
          summary = ""
          model = file(medaka_model).exists() ? "--medaka-model ./$medaka_model" : "--medaka-model $medaka_model"
      }
  def hd5_plugin_path = task.ext.hd5_plugin_path ? "export HDF5_PLUGIN_PATH=" + task.ext.hd5_plugin_path : "export HDF5_PLUGIN_PATH=/usr/local/lib/python3.6/site-packages/ont_fast5_api/vbz_plugin"
  """
      $hd5_plugin_path
  
      artic \\
          minion \\
          $args \\
          --threads $task.cpus \\
          --read-file $fastq \\
          --scheme-directory ./primer-schemes \\
          --scheme-version $version \\
          $model \\
          $fast5 \\
          $summary \\
          $scheme \\
          $prefix
  
      cat <<-END_VERSIONS > versions.yml
      "${task.process}":
          artic: \$(artic --version 2>&1 | sed 's/^.*artic //; s/ .*\$//')
      END_VERSIONS
      """

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

```

Execution script:

```
#!/bin/sh
#SBATCH --ntasks 1
#SBATCH --cpus-per-task 2
#SBATCH --mem 4G
#SBATCH --time 2:00:00
#SBATCH --partition middle_idx
#SBATCH --output 20220310_viralrecon.log
#SBATCH --chdir /scratch/bi/20220228_VIRALRECON-QC/ANALYSIS/20220307_ANALYSIS05_AMPLICONS_HUMAN_nanopore_v4

export NXF_OPTS="-Xms500M -Xmx4G"

nextflow run /scratch/bi/pipelines/nf-core-viralrecon-2.3/workflow/main.nf \
          -c /data/bi/pipelines/config_files/hpc_slurm.config \
	  -resume \
          --input samplesheet.csv \
          --outdir 20220310_viralrecon_mapping \
          --platform nanopore \
          --protocol amplicon \
	  --artic_scheme 'sars-cov-2' \
	  --genome 'MN908947.3' \
          --primer_set_version 4 \
	  --artic_minion_caller medaka \
	  --artic_minion_medaka_model r941_min_high_g360 \
	  --fastq_dir "/scratch/bi/20220228_VIRALRECON-QC/RAW/nanopore_articv4" \
	  --fasta "/data/bi/references/virus/2019-nCoV/genome/NC_045512.2.fasta" \
	  --gff "/data/bi/references/virus/2019-nCoV/genes/NC_045512.2.gff" \
	  --primer_bed "/data/bi/references/virus/2019-nCoV/amplicons/NC_045512.2/V4/nCoV-2019.artic.V4.scheme.bed" \
	  --kraken2_db "/data/bi/references/eukaria/homo_sapiens/hg38/UCSC/kraken2/kraken2_human.tar.gz" \
  --skip_assembly \
  --skip_nextclade

```

**Comment by Alema91** [comment]
I have tried again by using no compressed fastq because I thought that can be the problem but the same error goes on.

work folder to try:

```
/data/bi/scratch_tmp/bi/20220228_VIRALRECON-QC/ANALYSIS/20220307_ANALYSIS05_AMPLICONS_HUMAN_nanopore_v4/work/17/7f7a4779cc0b96ac00b154e4c4eda8
```

**Comment by Alema91** [comment]
Thanks to @saramonzon this issue was fixed in v2.4. Try again with newest version.

**Comment by Alema91** [comment]
Solved. Works fine with viralrecon v2.4.
